packages/flutter/lib/src/material/carousel.dart

```
@@ -482,6 +482,10 @@ class _CarouselViewState extends State<CarouselView> {
 
     return LayoutBuilder(
       builder: (BuildContext context, BoxConstraints constraints) {
+        if (constraints.maxWidth == 0 || constraints.maxHeight == 0) {
+          return const SizedBox();
+        }
```

Part of me is concerned that a guard clause returning an empty `SizedBox` could be seen as a "silent failure": if a developer sets up a CarouselView to have zero size constraints by accident, they might have an easier time figuring out what's wrong if they get an error message.

But if someone wants to animate the widget's size to & from zero, I think that's a valid use case that shouldn't result in an assertion error.

Overall I agree that this is a good change 👍

---

Swapping out with a sized box could have unintended results around the state of the carousel, the scrollable, and the state of the widget tree and how it is rebuilt. 

Consider this common case with zero constraints are introduced because the user has resized a window on desktop to be zero (we've run into this type of scenario before).